tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -128,55 +129,56 @@ def get_function_config(self, policy, queue_name=None):
+            DependencyManager.get_dependency_packages_list(modules, excluded_packages)
 
-        # MacOS
-        elif platform == "darwin":
-            raise NotImplementedError('Cannot package Azure Function in MacOS host OS, '
-                                      'please use linux.')
-        # Windows
-        elif platform == "win32":
-            raise NotImplementedError('Cannot package Azure Function in Windows host OS, '
-                                      'please use linux or WSL.')
+        if not DependencyManager.check_cache(cache_folder, wheels_install_folder, packages):
+            self.log.info("Cached packages not found or requirements were changed.")
+            # If cache check fails, wipe all previous wheels, installations etc
+            if os.path.exists(cache_folder):
+                self.log.info("Removing cache folder...")
+                shutil.rmtree(cache_folder)
 
-    def _update_perms_package(self):
-        os.chmod(self.pkg.path, 0o0644)
+            self.log.info("Preparing non binary wheels...")
+            DependencyManager.prepare_non_binary_wheels(non_binary_packages, wheels_folder)
 
-    def build(self, policy, queue_name=None, entry_point=None, extra_modules=None):
-        # Get dependencies for azure entry point
-        entry_point = entry_point or \
-            os.path.join(os.path.dirname(os.path.realpath(__file__)), 'entry.py')
-        modules, so_files = FunctionPackage._get_dependencies(entry_point)
+            self.log.info("Downloading wheels...")
+            DependencyManager.download_wheels(packages, wheels_folder)
 
-        # add all loaded modules
-        modules.discard('azure')
-        modules = modules.union({'c7n', 'c7n_azure', 'pkg_resources',
-                                 'knack', 'argcomplete', 'applicationinsights'})
-        if extra_modules:
-            modules = modules.union(extra_modules)
+            self.log.info("Installing wheels...")
+            DependencyManager.install_wheels(wheels_folder, wheels_install_folder)
 
-        self.pkg.add_modules(None, *modules)
+            self.log.info("Updating metadata file...")
+            DependencyManager.create_cache_metadata(cache_folder, wheels_install_folder, packages)
 
-        # adding azure manually
-        # we need to ignore the __init__.py of the azure namespace for packaging
-        # https://www.python.org/dev/peps/pep-0420/
-        self.pkg.add_modules(lambda f: f == 'azure/__init__.py', 'azure')
+        for root, _, files in os.walk(wheels_install_folder):
+            arc_prefix = os.path.relpath(root, wheels_install_folder)
+            for f in files:
+                dest_path = os.path.join(arc_prefix, f)
+
+                if f.endswith('.pyc') or f.endswith('.c'):
+                    continue
+                f_path = os.path.join(root, f)
+
+                self.pkg.add_file(f_path, dest_path)
+
+        self.pkg.add_modules(lambda f: ('\\cache\\' in f or '/cache/' in f),
```

Comment:
This is perfect as is, but it reminded me I recently learned about `os.path.normpath(path)`   - It just fixes the path automatically for the platform so you can do `os.path.normpath('/cache/')`  - or if you need to ensure the extraneous slashes remain intact then `os.path.sep` on either side I suppose.